x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -3109,6 +3109,15 @@ public void testFieldNamesAsCommands() throws Exception {
         }
     }
 
+    // [ and ( are used to trigger a double mode causing their symbol name (instead of text) to be used in error reporting
+    // this test checks that their are properly replaced in the error message
+    public void testPreserveParanthesis() {
+        // test for (
+        expectError("row a = 1 not in", "line 1:17: mismatched input '<EOF>' expecting '('");
```

Comment:
Should this be `row a = 1 | where a not in`?

Reply:
I was hoping, when I seen the title of this PR, that it would help with the "rogue" closing paren also #119025. But it doesn't, e.g. ```row a = 1 not in]``` or ```FROM test | eval x = [1, 2, 3]]``` .
```
java.util.EmptyStackException
  at org.antlr.v4.runtime.Lexer.popMode(Lexer.java:192)
  at org.antlr.v4.runtime.atn.LexerPopModeAction.execute(LexerPopModeAction.java:58)
  at org.antlr.v4.runtime.atn.LexerActionExecutor.execute(LexerActionExecutor.java:168)
...
```  
I only mention it since the use of modes, and the limitations of handling errors in ANTLR, are related, which might influence the course of action here.

Reply:
> Should this be row a = 1 | where a not in?
1. not really because row points to a value expression which is a bool expression. As the error indicates
2. I've added another test for it just in case.

Reply:
> I was hoping, when I seen the title of this PR

Reply:
> P.S. that should be a separate PR, since it needs backporting to the 8.x line vs this one that is targeting only 8.x.

 ... this one that is targeting only ~8.x~ 9.x.    Ah, yes, of course. Let's keep that separate then.